omegaconf/nodes.py

```diff
@@ -188,11 +191,9 @@ def __init__(
             ),
         )
 
-    def validate_and_convert(self, value: Any) -> Optional[int]:
+    def _validate_and_convert_impl(self, value: Any) -> Optional[int]:
         try:
-            if value is None:
-                val = None
-            elif type(value) in (str, int):
+            if type(value) in (str, int):
                 val = int(value)
             else:
                 raise ValueError()
```

Comment:
I would suggest to refactor this function's body like this:
```python
    def _validate_and_convert_impl(self, value: Any) -> int:
        if type(value) in (str, int):
            return int(value)
        else:
            raise ValidationError("Value '$VALUE' could not be converted to Integer")
```

Reply:
Your suggestion wouldn't raise the intended ValidationError if `value` is a string that can't be converted to `int`

Reply:
> Your suggestion wouldn't raise the intended ValidationError if `value` is a string that can't be converted to `int`

Oh, I see.